scripts/SANS/sans/command_interface/ISISCommandInterface.py

```diff
@@ -503,16 +502,46 @@ def SetPhiLimit(phimin, phimax, use_mirror=True):
     director.add_command(centre_command)
 
 
-def set_save(save_algorithms, save_as_zero_error_free):
+def set_save(
+    save_algorithms: Union[None, Dict] = None, save_as_zero_error_free: bool = False, output_mode: OutputMode = OutputMode.PUBLISH_TO_ADS
+) -> OutputMode:
     """
-    Mainly internally used by BatchMode. Provides the save settings.
+    Mainly used internally by WavRangeReduction. Prepares the save state on the director.
 
-    @param save_algorithms: A list of SaveType enums.
+    @param save_algorithms: A dict containing the name of the save algorithms as keys and extension as values.
     @param save_as_zero_error_free: True if a zero error correction should be performed.
+    @param output_mode: Decides if output_mode publishes to ads and saves to file or only one of the two when save_algorithms are valid
+    @return The OutputMode enum: PUBLISH_TO_ADS, SAVE_TO_FILE, BOTH
     """
-    save_command = NParameterCommand(command_id=NParameterCommandId.SAVE, values=[save_algorithms, save_as_zero_error_free])
+
+    if not save_algorithms:
```

Comment:
Now we've changed the logic of this function, does it make sense to move the logic regarding the output mode to `WavRangeReduction`?

```
    if saveAlgs:
        set_save(save_algorithms=saveAlgs, save_as_zero_error_free=save_as_zero_error_free)
        output_mode = output_mode.BOTH if (output_mode == OutputMode.PUBLISH_TO_ADS) else output_mode
```